go/mysql/datetime/parse.go

```diff
@@ -112,12 +120,9 @@ func parsetimeNoDelimiters(tp *timeparts, in string) (out string, ok bool) {
 	switch integral {
 	default:
 		// MySQL limits this to a numeric value that fits in a 32-bit unsigned integer.
-		i, _ := fastparse.ParseInt64(in[:integral], 10)
+		i, _ := fastparse.ParseUint64(in[:integral], 10)
 		if i > math.MaxUint32 {
-			return "", false
-		}
-		if i < -math.MaxUint32 {
```

Comment:
`i < -math.MaxUint32` is no longer needed to check?

Reply:
@systay I realized it's a useless check, since when we get here we already consumed any `-` at the beginning. So it can never be negative here, so hence the change to parse it as `uint` and drop this bit.